app/Components/MessageForm.js

```diff
@@ -18,24 +18,26 @@ class MessageForm extends React.Component {
     this.props.onMessageSubmit(message);  
     this.setState({ text: '' });
     this.refs.text.focus();
+    this.props.sendTypingState(false);
   }
 
   onChangeHandler(e) {
     if (e.target.value != "\n") {
       this.setState({ text : e.target.value });
+      this.props.sendTypingState(e.target.value.length > 0);
```

Comment:
Nice, this is a good way to make this work simply!  Some other options here are to use `_.throttle` or `_.debounce` to limit how often the call to update the typing state is made ([more info](https://css-tricks.com/debouncing-throttling-explained-examples/)).  Or to store a bit of state that has the timestamp of when the last change came in, and then use `setInterval` to check whether it's been `n` seconds since a keystroke.